test_parallel_reading_with_memory_limit is flaky

E.g. https://s3.amazonaws.com/clickhouse-test-reports/60669/75a84b4ab16b189c64deb1520344028bec91970b/integration_tests__aarch64__[5_6]//home/ubuntu/actions-runner/_work/_temp/test/output_dir/integration_run_parallel4_0.log

https://play.clickhouse.com/play?user=play#CldJVEgKICAgICd0ZXN0X3N0b3JhZ2VfczMvdGVzdC5weTo6dGVzdF9wYXJhbGxlbF9yZWFkaW5nX3dpdGhfbWVtb3J5X2xpbWl0JyBBUyBuYW1lX3N1YnN0ciwKICAgIDkwIEFTIGludGVydmFsX2RheXMsCiAgICAoJ1N0YXRlbGVzcyB0ZXN0cyAoYXNhbiknLCAnU3RhdGVsZXNzIHRlc3RzIChhZGRyZXNzKScsICdTdGF0ZWxlc3MgdGVzdHMgKGFkZHJlc3MsIGFjdGlvbnMpJykgQVMgYmFja3BvcnRfYW5kX3JlbGVhc2Vfc3BlY2lmaWNfY2hlY2tzClNFTEVDVAogICAgdG9TdGFydE9mRGF5KGNoZWNrX3N0YXJ0X3RpbWUpIEFTIGQsCiAgICBjb3VudCgpLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpCkZST00gY2hlY2tzCldIRVJFICgobm93KCkgLSB0b0ludGVydmFsRGF5KGludGVydmFsX2RheXMpKSA8PSBjaGVja19zdGFydF90aW1lKSBBTkQgKHB1bGxfcmVxdWVzdF9udW1iZXIgTk9UIElOICgKICAgIFNFTEVDVCBwdWxsX3JlcXVlc3RfbnVtYmVyIEFTIHBybgogICAgRlJPTSBjaGVja3MKICAgIFdIRVJFIChwcm4gIT0gMCkgQU5EICgobm93KCkgLSB0b0ludGVydmFsRGF5KGludGVydmFsX2RheXMpKSA8PSBjaGVja19zdGFydF90aW1lKSBBTkQgKGNoZWNrX25hbWUgSU4gKGJhY2twb3J0X2FuZF9yZWxlYXNlX3NwZWNpZmljX2NoZWNrcykpCikpIEFORCAocG9zaXRpb24odGVzdF9uYW1lLCBuYW1lX3N1YnN0cikgPiAwKSBBTkQgKHRlc3Rfc3RhdHVzIElOICgnRkFJTCcsICdFUlJPUicsICdGTEFLWScpKQpHUk9VUCBCWSBkCk9SREVSIEJZIGQgREVTQwo=

Clickhouse binary with hardcoded `max_block_size=8387` -> iIsue reproduced in my VM by running the integration tests (Standalone not yet reproduced)

```
========================================================================== short test summary info ==========================================================================
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[7-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[10-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[19-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[21-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[23-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[36-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[37-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[39-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[40-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[41-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[54-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[62-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[64-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[66-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[67-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[68-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[91-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[99-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
FAILED test_storage_s3/test.py::test_parallel_reading_with_memory_limit[100-100] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout: 0
================================================================= 19 failed, 81 passed in 744.40s (0:12:24) =================================================================
```

--- a/tests/integration/test_storage_s3/test.py
+++ b/tests/integration/test_storage_s3/test.py
@@ -1630,8 +1630,11 @@ def test_parallel_reading_with_memory_limit(started_cluster):
         f"insert into function s3('http://{started_cluster.minio_host}:{started_cluster.minio_port}/{bucket}/test_memory_limit.native') select * from numbers(1000000) SETTINGS s3_truncate_on_insert=1"
     )
 
+    # max_download_buffer_size=1048576 -> Trigger parallel http reads
+    # max_memory_usage=1000 -> Cause an exception in parallel read and server should not crash
+    # max_block_size=65409/max_untracked_memory=2097152 -> Ensure that query does not succeed
     result = instance.query_and_get_error(
-        f"select * from url('http://{started_cluster.minio_host}:{started_cluster.minio_port}/{bucket}/test_memory_limit.native') settings max_memory_usage=1000"
+        f"select * from url('http://{started_cluster.minio_host}:{started_cluster.minio_port}/{bucket}/test_memory_limit.native') settings max_memory_usage=1000,max_untracked_memory=2097152,max_block_size=65409,max_download_buffer_size=1048576"
     )
 
     assert "Query memory limit exceeded" in result

